app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -14,16 +14,31 @@ def google_oauth2
     end
   end
 
+  def facebook
+    if user.present?
+      user.accept_invitation! if invitation_token
+      flash[:notice] = I18n.t('devise.omniauth_callbacks.success',
+                              kind: 'Facebook')
+      sign_in_and_redirect @user, event: :authentication
+    else
+      redirect_to new_user_session_path, notice: t('omniauth.access_denied')
+    end
+  end
+
   private
 
   def user
-    @user ||= User.find_for_google_oauth2(request.env['omniauth.auth'])
+    @user ||= User.find_for_facebook(request.env['omniauth.auth'])
```

Comment:
```suggestion
    @user ||= User.find_for_oauth(request.env['omniauth.auth'])
```
I would consider refactring this to cater for both google and facebooth oauth in one fell swoop.